src/app/characters/character-list/character-list.component.ts

```diff
@@ -56,7 +56,13 @@ export class CharacterListComponent implements OnInit {
       switchMap(term => {
         this.showProgress = true;
         this.matSnackBarRef?.dismiss();
-        return this.characterService.getCharacters(term);
+
+        return this.characterService.getCharacters(term).pipe(
+          catchError(() => {
+            this.showProgress = false;
+            return EMPTY;
```

Comment:
```suggestion
         return of({ results: [], total: 0 });
```

@nikosanif thank you very much for this PR. What do you think about that? ☝️

Reply:
Hi @bampakoa! Sorry for the late response... 

I was thinking the same thing, but I realized that there is the case that we don't want to continue to the next `map` operator. The next PR (hopefully soon) will fix #21, where we are going to notify users about empty results. So, if we are going to return an empty list we will never be able to tell if there is an error or the API returned an empty list. 
Does it make sense to you?

Reply:
Absolutely! Thanks for the clarification 🙏